services/galley/src/Galley/API/Teams/Features.hs

```diff
@@ -632,18 +633,19 @@ getGuestLinkInternal ::
   GetFeatureInternalParam ->
   Sem r (Public.TeamFeatureStatus 'Public.WithLockStatus 'Public.TeamFeatureGuestLinks)
 getGuestLinkInternal = \case
-  Left _ -> getCfgDefault
+  Left _ -> do
+    traceM "internal param user"
+    getCfgDefault
   Right tid -> do
     cfgDefault <- getCfgDefault
     let defLockStatus = Public.tfwoapsLockStatus cfgDefault
     maybeFeatureStatus <- TeamFeatures.getFeatureStatusNoConfig @'Public.TeamFeatureGuestLinks tid
-    pure $ case (defLockStatus, maybeFeatureStatus) of
-      (Public.Unlocked, Just featureStatus) ->
+    pure $ case maybeFeatureStatus of
+      Just featureStatus ->
         Public.TeamFeatureStatusNoConfigAndLockStatus
           (Public.tfwoStatus featureStatus)
-          Public.Unlocked
-      (Public.Unlocked, Nothing) -> cfgDefault {Public.tfwoapsLockStatus = Public.Unlocked}
-      (Public.Locked, _) -> cfgDefault {Public.tfwoapsLockStatus = Public.Locked}
+          defLockStatus
+      Nothing -> cfgDefault
```

Comment:
I think this is a genuine fix, but it's still not completely correct.  Shouldn't the lock status in L647 come from cassandra, not the default?

(Wow, this is complicated, though.  Why is this so complicated?)

Reply:
(And have we been making the same mistake elsewhere?)